scripts/API.py

```diff
@@ -0,0 +1,438 @@
+# ***************************************************************************
+# * Authors:		Alberto García (alberto.garcia@cnb.csic.es)
+# *
+# *
+# * This program is free software; you can redistribute it and/or modify
+# * it under the terms of the GNU General Public License as published by
+# * the Free Software Foundation; either version 2 of the License, or
+# * (at your option) any later version.
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307 USA
+# *
+# * All comments concerning this program package may be sent to the
+# * e-mail address 'scipion@cnb.csic.es'
+# ***************************************************************************/
+
+"""
+Module containing all functions needed for the metric's API request.
+"""
+
+# General imports
+import re, hashlib, http.client, json
+
+# Self imports
+from typing import Optional, Tuple
+from utils import runJob
+
+API_URL = 'xmipp.i2pc.es/api/attempts/'
+
+def sendApiPOST(retCode: int = 0, XMIPP_VERSION:str = 'Unknow'):
+	"""
+	### Sends a POST request to Xmipp's metrics's API.
+
+	#### Params:
+	- retCode (int): Optional. Return code for the API request.
+	"""
+	try:
+		# Getting JSON data for curl command
+		bodyParams = __getJSON(retCode=retCode, XMIPP_VERSION=XMIPP_VERSION)
+
+		# Send API POST request if there were no errors
+		if bodyParams is not None:
+			# Set up the headers
+			headers = {"Content-type": "application/json"}
+
+			# Establish a connection
+			url = API_URL.split("/", maxsplit=1)
+			path = f"/{url[1]}"
+			url = url[0]
+			conn = http.client.HTTPConnection(url, timeout=4)
+
+			# Send the POST request
+			conn.request("POST", path, bodyParams, headers)
+			response = conn.getresponse()
+			status = response.status
+			# Close the connection
+			conn.close()
+			if status == 200 or status == 201:
+				return True
+			else:
+				print('-')
```

Comment:
Remove print or something more descriptive

Reply:
I would remove the if statement code block, as it looks like made for debugging purposes. The API function call should not produce any output and does not return any value (it should be virtually invisible to the user, and we don't need to do anything if the call succeeds/fails, so there's no point in providing a return value)

Reply:
The reason of this print is to have a footprint "virtually invisible" about a fails in the posting method

Reply:
I don't think this should be in production code. In case you want to track API failures, I would prefer to use the log with a more descriptive message.

Reply:
I agree with @oierlauzi, what is the point of printing something to the screen if we (developer side) won't be able to see it, as it will only be in the user's terminal. I agree that it should not be in production, and that a viable alternative could be adding a specific line to the log we could later trace.

Reply:
I liked the idea of having a tip-off that users couldn't interpret but we could. But I understand your position